src/components/Operations.js

```diff
@@ -0,0 +1,47 @@
+import React, {Component} from 'react';
+import {Link} from 'react-router-dom';
+
+class Operations extends Component {
+    constructor() {
+        super();
+        this.state = {
+            newAmount: 0,
+            newVendor: '',
+            newCategory: ''
+        }
+    }
+
+    updateAmount = (e) => this.setState({newAmount: parseInt(e.target.value)});  
```

Comment:
you do not need an input handler for each input. Instead give the input tag an attribute of name. Then grab that name using e.target.name
`handleInput = (e) => this.setState({[e.target.name]: parseInt(e.target.value)})`
`<input type = "number" name="newAmount" min = "0" id = "amount" onChange = {this.updateAmount}/>`